grep.md

```diff
@@ -0,0 +1,64 @@
+---
+title: Grep
+category: CLI
+layout: 2017/sheet
+updated: 2019-10-04
+---
+
+### Usage
+{: .-prime}
+
+```bash
+grep <options> pattern <file...>
+```
+
+### Matching Options
+
+```bash
+-e PATTERN, --regexp=PATTERN
+-f FILE, --file=FILE
+-i, --ignore-case
+-v, --invert-match
+-w, --word-regexp
+-x, --line-regexp
+```
+
+### Pattern Options
+
+```bash
+-E, --extended-regexp # extended regular expression
+-F, --fixed-strings   # list of fixed strings
+-G, --basic-regexp    # basic regular expression (default)
+-P, --perl-regexp     # perl compatible regular expression
+```
+
+### Output Options
+
+```bash
+-c, --count               # print the count of matching lines. suppresses normal output
+--color[=WHEN]            # applies color to the matches. WHEN is never, always, or auto
+-m NUM, --max-count=NUM   # stop reading after max count is reached
+-o, --only-matching       # only print the matched part of a line
+-q, --quiet, --silent
+-s, --no-messages         # suppress error messages about nonexistent or unreadable files
+```
+
+### Context Options
+
+```bash
+-A NUM, --after-context=NUM   # print NUM lines after a match
+-B NUM, --before-context=NUM  # print NUM lines before a match
+-C NUM, -NUM, --context=NUM   # print NUM lines before and after a match
```

Comment:
nitpick: maybe reorder A, B, C options to make the order following the context order:
- `--before`
- `--after`
- `--context`

or
- `--before`
- `--context`
- `--after`